packages/ui/src/components/sign-in/steps/choose-session.tsx

```diff
@@ -0,0 +1,175 @@
+import { useClerk } from '@clerk/clerk-react';
+import { cva } from 'cva';
+import { Button } from 'react-aria-components';
+
+import { useAppearance } from '~/hooks/use-appearance';
+import { useDisplayConfig } from '~/hooks/use-display-config';
+import { useLocalizations } from '~/hooks/use-localizations';
+import * as Card from '~/primitives/card';
+import * as Icon from '~/primitives/icon';
+
+function getInitials({
+  firstName,
+  lastName,
+  identifier,
+}: {
+  firstName?: string | null;
+  lastName?: string | null;
+  identifier?: string | null;
+}): string {
+  let initials: string = '';
+  if (firstName) {
+    initials += firstName[0].toUpperCase();
+  }
+  if (lastName) {
+    initials += lastName[0].toUpperCase();
+  }
+  if (!initials && identifier) {
+    initials += identifier[0].toUpperCase();
+  }
+  return initials;
+}
+
+function getTitleAndSubtitle({
+  firstName,
+  lastName,
+  identifier,
+}: {
+  firstName?: string | null;
+  lastName?: string | null;
+  identifier?: string | null;
+}) {
+  let title = '';
+  let subtitle = '';
+  if (firstName || lastName) {
+    title = `${firstName} ${lastName}`;
+    subtitle = identifier || '';
+  } else {
+    title = identifier || '';
+  }
+  return { title, subtitle };
+}
+
+const sessionAction = cva({
+  base: 'text-left text-gray-11 hover:bg-gray-2 data-[focus-visible]:bg-gray-2 flex w-full items-center gap-x-3 px-10 py-4 text-base outline-none [--session-icon-opacity:0] data-[hovered]:[--session-icon-opacity:1]',
+});
+
+export function SignInChooseSession() {
+  const clerk = useClerk();
+  const { t } = useLocalizations();
+  const { layout } = useAppearance();
+  // const renderDevModeWarning = useDevModeWarning();
+  const { branded } = useDisplayConfig();
+
+  const cardFooterProps = {
+    branded,
+    helpPageUrl: layout?.helpPageUrl,
+    privacyPageUrl: layout?.privacyPageUrl,
+    termsPageUrl: layout?.termsPageUrl,
+  };
+
+  const activeSessions = clerk.client.activeSessions;
```

Comment:
For now just pulling from the client for rendering, but will be replaced with elements once that work is completed.